### PR TITLE
Example of purpose built dashboard queries

### PR DESCRIPTION
**Pull request by wssheldon**
Screenshot 2023-05-25 at 12 47 32 PM
Screenshot 2023-05-25 at 12 47 27 PM
Screenshot 2023-05-25 at 12 49 33 PM

Crude example, this approach helps w/ performance a lot, and has added benefit of each card loading async without waiting for main one. We'd need to add filtering support.

### CODE REVIEW

File: src/dispatch/case/service.py [hunk 1]
```diff
@@ -80,6 +83,16 @@ def get_all_by_status(*, db_session, project_id: int, status: str) -> List[Optio
     )
 
 
+def get_count_of_case_types(*, db_session) -> list[tuple[str, int]]:
```
Comment (wssheldon) [inline src/dispatch/case/service.py:86]: should be `case/type` dir

File: src/dispatch/case/views.py [hunk 2]
```diff
@@ -246,3 +246,16 @@ def delete_case(
                 }
             ],
         ) from None
+
+
+@router.get(
```
Comment (wssheldon) [inline src/dispatch/case/views.py:251]: should be in `case/type`

File: src/dispatch/static/dispatch/src/dashboard/case/CaseTypeBarChartCard.vue [hunk 3]
```diff
@@ -40,17 +40,15 @@ export default {
...
   created: function () {
-    CaseTypeApi.getAll({ itemsPerPage: -1 }).then((response) => {
-      this.types = map(
-        filter(response.data.items, function (item) {
-          return !item.exclude_from_metrics
-        }),
-        "name"
-      )
+    CaseApi.getTypeCount().then((response) => {
+      console.log(response)
```
Comment (wssheldon) [inline src/dispatch/static/dispatch/src/dashboard/case/CaseTypeBarChartCard.vue:49]:
```suggestion
```

File: src/dispatch/static/dispatch/src/dashboard/case/CaseTypeBarChartCard.vue [hunk 4]
```diff
@@ -40,17 +40,15 @@ export default {
...
 
   created: function () {
-    CaseTypeApi.getAll({ itemsPerPage: -1 }).then((response) => {
-      this.types = map(
-        filter(response.data.items, function (item) {
-          return !item.exclude_from_metrics
-        }),
-        "name"
-      )
+    CaseApi.getTypeCount().then((response) => {
```
Comment (wssheldon) [inline src/dispatch/static/dispatch/src/dashboard/case/CaseTypeBarChartCard.vue:48]:
```suggestion
    CaseTypeApi.getTypeCount().then((response) => {
```

would need to move method to apply

File: src/dispatch/case/service.py (hunk 1 above)
Comment (mvilanova) [reply src/dispatch/case/service.py:86]: Could we generalize and parameterize this function to accept the objects for which we want the counts, so we don't have to write one for each data type?

Comment (wssheldon) [reply src/dispatch/case/service.py:86]: Yeah potentially, it may get tricky if we need to join across tables.

**Pull request closed without merging**